format_test.go

```diff
@@ -52,6 +52,11 @@ var formatTests = []struct {
 		i: pointer(30),
 		s: pointer(0),
 	}, "last", 1551342600},
+	{"american", formatsMap["american"], &result{
+		y: pointer(2019),
+		m: pointer(10),
```

Comment:
Months are zero-indexed, so they range from 0 (January) to 11 (December). If you change this to 9 it should work.
```suggestion
		m: pointer(9),
```